components/screens/ObservationsScreen.tsx

```diff
@@ -46,6 +60,42 @@ const ObservationsListScreen = ({route}: NativeStackScreenProps<ObservationsStac
   );
 };
 
+/**
+ * Return the previous screen from the route stack.
+ */
+const lastScreen = (state: StackNavigationState<ObservationsStackParamList>) => state.routes[state.index - 1];
```

Comment:
nit: for really simple things & especially as they're just called once, I prefer to inline this